torch/csrc/jit/type.h

```diff
@@ -295,7 +316,7 @@ struct TORCH_API CompleteTensorType : public TensorType {
   CompleteTensorType(at::ScalarType scalar_type, int device, at::IntList sizes)
     : CompleteTensorType(scalar_type, device, sizes, CompleteTensorType::contiguousStridesOf(sizes)) {}
   CompleteTensorType(at::ScalarType scalar_type, int device, at::IntList sizes, at::IntList strides)
-    : TensorType(scalar_type, device, sizes.size(), TypeKind::CompleteTensorType)
+    : TensorType(scalar_type, device, sizes.size(), true, TypeKind::CompleteTensorType)
```

Comment:
Why are complete tensor types always requires_grad=True?

Reply:
We don't really use those in many places so I didn't bother with adding the argument to this constructor. The important part to understand about the `requires_grad` flags of types is that they are an over-approximation. Effectively we're trying to prove as much as we can that some values *don't* require grad, and otherwise we need to conservatively consider them as differentiable (because some argument paths *might* cause them to be).

I just realized that we might not even need `might_introduce_grad` in `GraphExecutor`. Return types of `PythonOp`s will always be `DynamicType`, and it has `requires_grad` always set to `true` (because we can't prove anything about it!).

Reply:
Yeah, it would be nice to get rid of that. I feel like CompleteTensorType should at least have it as an optional argument, because if it is not there someone writing new code will not even considering passing along requires_grad information even when they have it, which could lead to unintended de-optimization.

Reply:
Ok, I can add it.